library/core/src/num/mod.rs

```diff
@@ -822,34 +822,39 @@ fn from_str_radix<T: FromStrRadixHelper>(src: &str, radix: u32) -> Result<T, Par
         radix
     );
 
-    if src.is_empty() {
-        return Err(PIE { kind: Empty });
-    }
-
     let is_signed_ty = T::from_u32(0) > T::min_value();
 
     // all valid digits are ascii, so we will just iterate over the utf8 bytes
     // and cast them to chars. .to_digit() will safely return None for anything
     // other than a valid ascii digit for the given radix, including the first-byte
     // of multi-byte sequences
-    let src = src.as_bytes();
-
-    let (is_positive, digits) = match src[0] {
-        b'+' | b'-' if src[1..].is_empty() => {
-            return Err(PIE { kind: InvalidDigit });
+    let mut src = src.as_bytes().iter().enumerate().peekable();
+
+    let first_digit = src.peek().ok_or(PIE { kind: Empty })?.1;
+
+    let (is_positive, digits) = match first_digit {
+        b'+' | b'-' => {
+            src.next();
+            if src.peek().is_none() {
+                return Err(PIE { kind: InvalidDigit(0) });
```

Comment:
Why should `0` be there when it is `is_none`?

Reply:
It is to handle the strings "+" and "-". Those single symbols are considered invalid when followed by nothing else. 

`test_parse::<i8>("-", Err(IntErrorKind::InvalidDigit(0)));`